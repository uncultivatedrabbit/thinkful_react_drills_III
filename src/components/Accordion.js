import React, { Component } from "react";

export default class Accordion extends Component {
  static defaultProps = { sections: [] };
  constructor(props) {
    super(props);
    this.state = {
      currentAccordion: null,
    };
  }

  handleClick = i => {
    this.setState({ currentAccordion: i });
  };

  renderContent(section, i, currentAccordion) {
    return (
      <li key={i}>
        <button type="button" onClick={() => this.handleClick(i)}>
          {section.title}
        </button>
        {currentAccordion === i && <p>{section.content}</p>}
      </li>
    );
  }

  displayContent() {
    const { sections } = this.props;
    const { currentAccordion } = this.state;
    return sections.map((section, i) =>
      this.renderContent(section, i, currentAccordion)
    );
  }
  render() {
    return (
      <>
        <ul>{this.displayContent()}</ul>
      </>
    );
  }
}
