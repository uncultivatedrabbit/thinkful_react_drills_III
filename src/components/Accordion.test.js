import React from "react";
import toJson from "enzyme-to-json";
import { shallow} from "enzyme";
import Accordion from "./Accordion";


const testSections = [
  {
    title: "Section 1",
    content: "Lorem ipsum dolor sit amet consectetur adipisicing elit.",
  },
  {
    title: "Section 2",
    content:
      "Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!",
  },
  {
    title: "Section 3",
    content:
      "Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?",
  },
];

describe("Accordion", () => {
  it("renders empty given no Accordion sections without errs", () => {
    const wrapper = shallow(<Accordion />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders no sections by default", () => {
    const wrapper = shallow(<Accordion sections={testSections}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('opens clicked sections',() => {
    const wrapper = shallow(<Accordion sections={testSections}/>)
    wrapper.find('button').at(1).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('only opens one section at a time', () => {
    const wrapper = shallow(<Accordion sections={testSections} />)
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('button').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
});
