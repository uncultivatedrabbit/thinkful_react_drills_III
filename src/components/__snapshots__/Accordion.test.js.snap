// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion only opens one section at a time 1`] = `
<Fragment>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 1
      </button>
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 2
      </button>
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 3
      </button>
      <p>
        Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
      </p>
    </li>
  </ul>
</Fragment>
`;

exports[`Accordion opens clicked sections 1`] = `
<Fragment>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 1
      </button>
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 2
      </button>
      <p>
        Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
      </p>
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 3
      </button>
    </li>
  </ul>
</Fragment>
`;

exports[`Accordion renders empty given no Accordion sections without errs 1`] = `
<Fragment>
  <ul />
</Fragment>
`;

exports[`Accordion renders no sections by default 1`] = `
<Fragment>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 1
      </button>
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 2
      </button>
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 3
      </button>
    </li>
  </ul>
</Fragment>
`;
